mssql.go

```diff
@@ -1145,6 +1147,10 @@ func (rc *Rowsq) Next(dest []driver.Value) error {
 			if tok == nil {
 				return io.EOF
 			} else {
+				switch tokdata := tok.(type) {
+				case doneInProcStruct:
+					tok = (doneStruct)(tokdata)
```

Comment:
is there a more idiomatic way to combine the handling of `doneInProcStruct` and `doneStruct`?